packages/api-client-core/src/support.ts

```diff
@@ -588,6 +588,9 @@ export const disambiguateActionVariables = (action: AnyActionFunction, variables
 
   let newVariables: Record<string, any>;
 
+  // for backwards compatibilty, actions without the operatesWithRecordIdentity metadata should extract the id from the variables
+  const shouldExtractId = action.operatesWithRecordIdentity ?? true;
```

Comment:
if we leave `operatesWithRecordIdentity` as optional, then this will make sure that by default we extract id as we currently do; I think before upsert this is fine

Reply:
Game for adding a new property, but could we also look at `action.variables` to see if there's an id arg at the root and avoid juggling a new prop? i.e.,

```
const shouldExtractId = "id" in action.variables;
```

Reply:
I think this doesn't work for bulk actions where the top level variables are `inputs`; but we still disambiguate in bulk